source/documentation/deploying_services/route_services.md

```diff
@@ -0,0 +1,108 @@
+
+   ``502 Bad Gateway: Registered endpoint failed to handle the request.``
+
+If you get this error, double check that the endpoint is working and reachable from the platform, that it is using a valid SSL certificate, that responds timely and that it implements the [Route Service Protocol](/#implementing-a-route-service).
+
+### Implementing a Route Service
+
+A Route Service can be implemented as a HTTPS application that will receive all requests for the associated route from the Cloud Foundry router. The request will contain the following additional headers:
+
+- `X-CF-Forwarded-Url` header contains the URL of the application route. The route service should forward the request to this URL.
+- `X-CF-Proxy-Signature` and `X-CF-Proxy-Metadata`: Signature and metadata used by the platform route itself to validate the request sent by the route service.
+  The route service must always include these headers.
+
+The Route Services must proxy back the request to the application route defined in header `X-CF-Forwarded-Url` within 60 seconds, otherwise the request will be refused. In addition, the total time to process the request and send a response back must be within 900 seconds.
+
+![Route Service request life cycle](images/route-service.png)
+
+You can refer to the [Cloud Foundry documentation](https://docs.cloudfoundry.org/services/route-services.html#service-instance-responsibilities), [example route services](https://docs.cloudfoundry.org/services/route-services.html#examples), and [tutorial](https://docs.cloudfoundry.org/services/route-services.html#tutorial) to learn more about how to implement route services.
+
+### Example: Route Service to add authentication
+
+In the following example we will add a Route Service to provide basic HTTP authentication to the bound routes. This can be used to restrict access to a beta application.
+
+An example of such a Route Service application code can be found in [cf basic auth route service](https://github.com/alext/cf_basic_auth_route_service).
+Please note this is a proof-of-concept and is *not intended to run in production*.
+
+We will deploy it as an application in the platform itself. Then we will bind this Route Service to a deployed application accessible via https://myapp.cloudapps.digital.
+
+1. Deploy the route service as [any other other application](/#deploying-apps).
+   Do not start it yet, as we need to define the user and password.
+
+    ```
+    git clone https://github.com/alext/cf_basic_auth_route_service
+    cd cf_basic_auth_route_service
+    cf push my-basic-auth-service-app --no-start
+    ```
+
+   Note: You might want to change `my-basic-auth-service-app` with a different name to avoid conflicts with other tenants apps.
+
+2. For this example, the user and password can be defined with the environment variables `USER_USERNAME` and `AUTH_PASSWORD`, which we must set. Start the Route Service application once set:
```

Comment:
I was a little confused by this section - how does the Tenant set usernames and passwords? Can they do it or should they contact us and we do it? It the latter, we should say so. Also, where we say 'Start the Route Service application once set:'... should we define what is set? Is it username and password?